vividus-plugin-web-app-to-rest-api/src/main/java/org/vividus/bdd/steps/integration/ResourceCheckSteps.java

```diff
@@ -119,24 +119,38 @@ private ResourceValidation validate(ResourceValidation r)
         return HtmlUtils.getElements(html, cssSelector).parallelStream();
     }
 
+    private boolean isSourceAttributeEmpty(Element element)
+    {
+        boolean isSourceAttributeEmpty = getElementSourceAttribute(element).isEmpty();
+        if (isSourceAttributeEmpty)
+        {
+            softAssert.recordFailedAssertion(
```

Comment:
shouldn't we adjust resource checking logic depending on an element being checked? for examples 'a' element is allowed to be without href attr whereas 'img' isn't
https://www.w3.org/TR/html52/semantics-embedded-content.html#the-img-element
https://www.w3.org/TR/html52/textlevel-semantics.html#the-a-element

Reply:
I suppose it doesn't matter, the only one restriction is: an element must have src or href

Reply:
I meant why should we fail during resource checking of links if it's not prohibited by spec to have <a> without href?

Reply:
it causes StringIndexOutOfBoundsException otherwise

Reply:
thanks for updating the docs :)